### PR TITLE
Throw error when `creationRoleProvider` is used in MLC programs

### PR DESCRIPTION
**Pull request by rquitales**
### Proposed changes

The `creationRoleProvider` option in the nodejs version of this provider allows passing a Pulumi AWS provider. However, for MLC programs, this isn't supported since only the JSON serialized provider is passed, not a fully initialized instance.

A user attempting to use this option will encounter a non-helpful error message: `Detail="provider.getPackage is not a function")`

This PR updates the provider to determine if the provider is being passed, and if so, returns a more helpful message on how to configure the EKS cluster settings.

### Related issues (optional)

Closes: #1113

### CODE REVIEW

File: examples/cluster-go/step2/main.go [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+		if err != nil {
+			return err
+		}
+
+		////////////////////////////////////////////////
+		///     CreationRoleProvider should fail     ///
+		////////////////////////////////////////////////
+
+		// Create AWS provider
+		awsProv, err := aws.NewProvider(ctx, "aws", nil)
```
Comment (rquitales) [inline examples/cluster-go/step2/main.go:103]: The new resource being added in step 2.

File: nodejs/eks/cluster.ts [hunk 2]
```diff
@@ -1905,6 +1905,12 @@ export class ClusterInternal extends pulumi.ComponentResource {
 
         super(type, name, args, opts);
 
+        if (args?.creationRoleProvider?.provider) {
+            throw new Error(
+                "The `creationRoleProvider.provider` option is not supported in non-nodejs Pulumi programs. Please use the `providerCredentialOpts` option instead.",
+            );
+        }
```
Comment (iwahbe) [inline nodejs/eks/cluster.ts:1912]: Is there an issue for serializing resources in nodejs MLCs in Pulumi? If so, can we link it. If not, can we open the feature request and link it?

**Comment by t0yv0** [comment]
> However, for MLC programs, this isn't supported since only the JSON serialized provider is passed, not a fully initialized instance.

I am a little surprised at that, since we have resource references that represent resources on the wire.  I was working on the feature here that allows a component to configure a Provider and then the program gets it through a method and uses that Provider to configure more resources: 

https://github.com/pulumi/pulumi/blob/master/tests/integration/construct_component_configure_provider/nodejs/index.ts that

**Comment by t0yv0** [comment]
Is this this one https://www.pulumi.com/registry/packages/eks/api-docs/clustercreationroleprovider/ 

https://www.pulumi.com/registry/packages/eks/api-docs/cluster/#creationroleprovider_nodejs

**Comment by t0yv0** [comment]
I think this change is not unreasonable but I'd double-check if other languages work. Presumably Node works. C# does not work. Go and Python may just work actually? If not let's indeed ticket a feature in p/p - there's very likely a bug of some sort because resource references are supposed to travel over the wire just fine, including provider references; perhaps the gap is C# specific .

**Comment by rquitales** [comment]
> I think this change is not unreasonable but I'd double-check if other languages work. Presumably Node works. C# does not work. Go and Python may just work actually? If not let's indeed ticket a feature in p/p - there's very likely a bug of some sort because resource references are supposed to travel over the wire just fine, including provider references; perhaps the gap is C# specific .

I've confirmed that this issue affects all go, python and c#. I didn't get a chance to test java, but I would assume it should be the same outcome. Unfortunately, using the [ClusterCreationRoleProvider](https://www.pulumi.com/registry/packages/eks/api-docs/clustercreationroleprovider/) also doesn't work, since it would be a type mismatch resulting in `unknown resource type eks:index:ClusterCreationRoleProvider`.

**Review by t0yv0 — APPROVED** [review]
This is the right fix then thank you for confirming. Looking a bit to see how can we extract a good ticket for Pulumi Core here, what's going wrong...

**Pull request merged**